internal/pkg/manifest/svc.go

```diff
@@ -442,3 +443,90 @@ func (s *stringOrFromCFN) UnmarshalYAML(value *yaml.Node) error {
 	}
 	return nil
 }
+
+func (cfg ImageWithPortAndHealthcheck) exposedPorts(workloadName string) []ExposedPort {
+	// Read `image.port`
+	if cfg.Port == nil {
+		return nil
+	}
+	return []ExposedPort{
+		{
+			Port:          aws.Uint16Value(cfg.Port),
+			Protocol:      "tcp",
+			ContainerName: workloadName,
+		},
+	}
+
+}
+
+func (cfg ImageWithHealthcheckAndOptionalPort) exposedPorts(workloadName string) []ExposedPort {
+	// Read `image.port`
+	if cfg.Port == nil {
+		return nil
+	}
+	return []ExposedPort{
+		{
+			Port:          aws.Uint16Value(cfg.Port),
+			Protocol:      "tcp",
+			ContainerName: workloadName,
+		},
+	}
+}
+
+func (rr RoutingRuleConfiguration) exposedPorts(exposedPorts []ExposedPort, workloadName string) []ExposedPort {
```

Comment:
Since this is really the only one that's particularly tricky, I think it's worthwhile adding a comment explaining why this method takes an extra parameter.
```suggestion
// exportPorts returns any new ports that should be exposed given the load balancer configuration that's not part of the existing containerPorts.
func (rr RoutingRuleConfiguration) exposedPorts(containerPorts []ExposedPort, defaultContainerName string) []ExposedPort {
```